vowpalwabbit/parse_primitives.cc

```diff
@@ -151,4 +155,45 @@ std::vector<std::string> split_command_line(const std::string& cmd_line)
 {
   return split_impl(cmd_line.begin(), cmd_line.end());
 }
+
+std::string escape_string(VW::string_view str)
+{
+  std::stringstream ss;
+  for (char c : str)
+  {
+    switch (c)
+    {
+      case '\n':
+        ss << R"(\n)";
+        break;
+      case '\t':
+        ss << R"(\t)";
+        break;
+      case '\\':
+        ss << R"(\)";
+        break;
+      case '"':
+        ss << R"(\")";
+        break;
+      case '\'':
+        ss << R"(\')";
+        break;
+      default:
+        ss << c;
+        break;
+    }
+  }
+  return ss.str();
+}
+
+bool contains_escapeable_chars(VW::string_view str)
+{
+  const std::set<char> escape_chars = {'\n', '\t', '\\', '"', '\''};
```

Comment:
`\r`?

Reply:
Yeah I see what you are saying. Having a `\n` or a `\t` in a value of a cli arg is very particular

Reply:
Going to reduce the scope of escaping to just \ ' "